### PR TITLE
Create Challenges Completion Progress Widget

### LINKED ISSUE #32: Create Challenges Completion Progress Widget
## Description
Develop a widget component to display the total percentage of challenges completed out of all available challenges, including a progress bar and the date of the next challenge delivery.

Single bar

## Motivation
To provide users with an overview of their progress across all available challenges and highlight upcoming challenges, we need to create a widget component that visually represents the completion rate and displays the next challenge delivery date.

## Useful Links
[Figma Design](https://www.figma.com/design/xUlM0BIpE50NX1YItvSdGu/Escopo-projeto?node-id=295-4339&t=W3JcLYGYsyrLSRju-0)

## Acceptance Criteria
- [ ] The data structure and props required for the component are defined
- [ ] The component displays the total percentage of challenges completed out of all available challenges
- [ ] The component includes a progress bar to visually represent the completion percentage
- [ ] The component displays the date of the next challenge delivery
- [ ] The component is responsive and looks good on different screen sizes
- [ ] The challenge completion progress widget component is styled according to the design specifications

### PR DESCRIPTION
**Pull request by roberto-juniorjp**
## Description
closes #32
Added a challenges progress completion widget. For that to work out, i also installed tremor plugin along with all its dependencies, as well as update the lockfiles and tailwind.config.ts files to comport the new plugin.

[tremor: ](https://www.tremor.so/docs/getting-started/installation)

## Todo
- [x]  The data structure and props required for the component are defined
- [x] The component displays the total percentage of challenges completed out of all available challenges
- [x] The component includes a progress bar to visually represent the completion percentage
- [x] The component displays the date of the next challenge delivery
- [x] The component is responsive and looks good on different screen sizes
- [x] The challenge completion progress widget component is styled according to the design specifications

### CODE REVIEW

File: src/components/challenges-completion.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import { BarProgress } from './ui/graphElements';
+import Widget from './ui/widget';
+
+interface ChallengesCompletionProps {
+  totalChallengePercentage: number;
+  nextChallengeDate: string;
+}
+
+const ChallengesCompletion = (
+  challengesCompletionProps: ChallengesCompletionProps
```
Comment (RO-HSA) [inline src/components/challenges-completion.tsx]: Destructure the props could be a good idea here

```suggestion
  { totalChallengePercentage, nextChallengeDate }: ChallengesCompletionProps
```

File: src/components/challenges-completion.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            <div className='col-span-2 items-center justify-center'>
+              <BarProgress
+                totalChallengePercentage={30}
+                barLabel={'Total de desafios completos'}
+              />
+            </div>
+            <div className='flex-cols flex items-center justify-center'>
+              <h3 className='text-5xl'>
+                {challengesCompletionProps.totalChallengePercentage}%
+              </h3>
```
Comment (RO-HSA) [inline src/components/challenges-completion.tsx]:
```suggestion
                {totalChallengePercentage}%
```

File: src/components/challenges-completion.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            </div>
+          </div>
+        </div>
+        <div className='grid grid-cols-3'>
+          <div className='items- col-span-2 flex items-center font-bold'>
+            <p className='text-brand-300'>
+              Data da próxima entrega:{' '}
+              <span className='text-brand-400'>
+                {challengesCompletionProps.nextChallengeDate}
```
Comment (RO-HSA) [inline src/components/challenges-completion.tsx]:
```suggestion
                {nextChallengeDate}
```

File: src/components/user-level.tsx [hunk 4]
```diff
@@ -6,28 +6,32 @@ import { CircularProgress } from '@/components/ui/graphElements';
...
+interface UserLevelProps {
+  nextLevelPercentage: number;
   level: number;
 }
 
-const UserLevel = (scoreProps: ScoreProps) => {
-  const [score] = useState<number>(scoreProps.score);
-  const [level] = useState<number>(scoreProps.level);
+const UserLevel = (userLevelProps: UserLevelProps) => {
+  const [nextLevelPercentage] = useState<number>(
```
Comment (RO-HSA) [inline src/components/user-level.tsx]: Destructure the object could be a good idea here

```suggestion
const UserLevel = ({ nextLevelPercentage, level }: UserLevelProps) => {
```

File: src/components/user-level.tsx [hunk 5]
```diff
@@ -6,28 +6,32 @@ import { CircularProgress } from '@/components/ui/graphElements';
...
+    userLevelProps.nextLevelPercentage
+  );
```
Comment (RO-HSA) [inline src/components/user-level.tsx]:
```suggestion
    nextLevelPercentage
```

File: src/components/user-level.tsx [hunk 6]
```diff
@@ -6,28 +6,32 @@ import { CircularProgress } from '@/components/ui/graphElements';
...
         <div className='h-full w-full content-around'>
           <div className='flex flex-col items-center justify-center border-r-2 border-brand-400 py-12'>
-            <h2 className='text-4xl font-bold uppercase'>Bichinho</h2>
-            <h2 className='text-4xl font-bold uppercase'>
-              Nível <span className='text-brand-500'>{level}</span>
+            <h2 className='text-4xl font-bold'>Bichinho</h2>
+            <h2 className='text-4xl font-bold'>
+              Nível{' '}
+              <span className='text-brand-500'>{userLevelProps.level}</span>
             </h2>
```
Comment (RO-HSA) [inline src/components/user-level.tsx:21]:
```suggestion
              <span className='text-brand-500'>{level}</span>
```

File: src/components/user-level.tsx (hunk 4 above)
Comment (RO-HSA) [inline src/components/user-level.tsx]:
```suggestion
  const [nextLevel] = useState<number>(
```

File: src/components/user-level.tsx [hunk 7]
```diff
@@ -6,28 +6,32 @@ import { CircularProgress } from '@/components/ui/graphElements';
...
           </div>
         </div>
         <div className='flex items-center justify-center'>
-          <CircularProgress>{score}</CircularProgress>
+          <CircularProgress
+            nextLevelPercentage={nextLevelPercentage}
+          ></CircularProgress>
```
Comment (RO-HSA) [inline src/components/user-level.tsx:27]:
```suggestion
            nextLevelPercentage={nextLevel}
```

**Review by RO-HSA — CHANGES_REQUESTED** [review]
Just a few things to change

**Comment by roberto-juniorjp** [comment]
@RO-HSA adjustments made. Please re-review.

**Review by RO-HSA — APPROVED** [review]
LGTM

**Pull request merged**